File: VB/NetFramework/net48/03_Focus/Tab.vb
Imports GrapeCity.Win.Editors
Imports System.Windows.Forms

Namespace _03_Focus
    Partial Public Class Tab
        Inherits InputManWin13_Demo.DemoBase_settings
        Public Sub New()
            InitializeComponent()
            InitializeForm()
        End Sub

        Private Sub InitializeForm()
            ' ***** 初期値
            gcComboBox1.SelectedIndex = If((gcTextBox1.AcceptsTab = True), 0, 1)
            gcComboBox2.SelectedIndex = Convert.ToInt32(gcMask1.TabAction)

            ' xmlファイルからデータ取得(GcMaskedComboBox)
            Dim filePath As System.String = Application.StartupPath
            Me.DataSet1.ReadXml(filePath & "\Resources\pref.xml")
            GcMaskedComboBox1.DataSource = DataSet1.Tables(0)
            GcMaskedComboBox1.SelectedIndex = -1

            For i As Integer = 1 To GcMaskedComboBox1.ListColumns.Count - 1
                GcMaskedComboBox1.ListColumns(i).AutoWidth = True
            Next
        End Sub

        Private Sub gcComboBox1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcComboBox1.SelectedIndexChanged
            ' AcceptsTabプロパティ
            gcTextBox1.AcceptsTab = If((gcComboBox1.SelectedIndex = 0), True, False)
        End Sub

        Private Sub gcComboBox2_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcComboBox2.SelectedIndexChanged
            ' TabActionプロパティ
            gcMask1.TabAction = CType(gcComboBox2.SelectedIndex, TabAction)
            gcDateTime1.TabAction = CType(gcComboBox2.SelectedIndex, TabAction)
            gcDate1.TabAction = CType(gcComboBox2.SelectedIndex, TabAction)
            gcTime1.TabAction = CType(gcComboBox2.SelectedIndex, TabAction)
            gcTimeSpan1.TabAction = CType(gcComboBox2.SelectedIndex, TabAction)
            GcMaskedComboBox1.TabAction = CType(gcComboBox2.SelectedIndex, TabAction)
        End Sub
    End Class
End Namespace

File: CS/NetFramework/net48/03_Focus/Tab.cs
using GrapeCity.Win.Editors;
using System;
using System.Windows.Forms;

namespace InputManWin13_Demo._03_Focus
{
    public partial class Tab : InputManWin13_Demo.DemoBase_settings
    {
        public Tab()
        {
            InitializeComponent();
            InitializeForm();
        }

        private void InitializeForm()
        {
            // ***** 初期設定
            gcComboBox1.SelectedIndexChanged += new EventHandler(gcComboBox1_SelectedIndexChanged);
            gcComboBox2.SelectedIndexChanged += new EventHandler(gcComboBox2_SelectedIndexChanged);

            // ***** 初期値
            gcComboBox1.SelectedIndex = (gcTextBox1.AcceptsTab == true) ? 0 : 1;
            gcComboBox2.SelectedIndex = Convert.ToInt32(gcMask1.TabAction);

            // xmlファイルからデータ取得（gcMaskedComboBox）
            System.String filePath = Application.StartupPath;
            this.dataSet1.ReadXml(filePath + "\\Resources\\pref.xml");
            gcMaskedComboBox1.DataSource = dataSet1.Tables[0];
            gcMaskedComboBox1.SelectedIndex = -1;

            for (int i = 1; i < gcMaskedComboBox1.ListColumns.Count; i++)
            {
                gcMaskedComboBox1.ListColumns[i].AutoWidth = true;
            }
        }

        private void gcComboBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            // AcceptsTabプロパティ
            gcTextBox1.AcceptsTab = (gcComboBox1.SelectedIndex == 0) ? true : false;
        }

        private void gcComboBox2_SelectedIndexChanged(object sender, EventArgs e)
        {
            // TabActionプロパティ
            gcMask1.TabAction = (TabAction)gcComboBox2.SelectedIndex;
            gcDateTime1.TabAction = (TabAction)gcComboBox2.SelectedIndex;
            gcDate1.TabAction = (TabAction)gcComboBox2.SelectedIndex;
            gcTime1.TabAction = (TabAction)gcComboBox2.SelectedIndex;
            gcTimeSpan1.TabAction = (TabAction)gcComboBox2.SelectedIndex;
            gcMaskedComboBox1.TabAction = (TabAction)gcComboBox2.SelectedIndex;
        }
    }
}